wpilibc/src/main/native/cpp/PIDController.cpp

```diff
@@ -498,9 +520,9 @@ void PIDController::SetToleranceBuffer(int bufLength) {
  * This will return false until at least one input value has been computed.
  */
 bool PIDController::OnTarget() const {
-  double error = GetError();
-
+  if (StatusIsFatal()) return false;
   std::lock_guard<wpi::mutex> lock(m_mutex);
+  double error = GetError();
```

Comment:
`double error = GetError();` belongs above the std::lock_guard. It'll deadlock with where it is now because GetError() calls functions that also lock the non-recursive mutex.

Reply:
Got it, nice catch